File: CarteiraVacinaDigital.Model/Entities/PacientVaccine.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace CarteiraVacinaDigital.Model.Entities
{
    public class PacientVaccine
    {
        public int ID { get; set; }
        public virtual int PacientID { get; set; }
        public virtual int VaccineID { get; set; }
        public DateTime DateVaccined { get; set; }
        public bool UniqueDose { get; set; }
        public int? Dose { get; set; }
        [NotMapped]
        public virtual Pacient Pacient { get; set; }
        [NotMapped]
        public virtual Vaccine Vaccine { get; set; }
    }
}

File: CarteiraVacinaDigital.Api/ClientApp/src/app/core/model/pacientVaccine.ts
import { Pacient } from "./pacient";
import { Vaccine } from "./vaccine";

export class PacientVaccine {
    id: number;
    PacientId: number;
    VaccineId: number;
    DateVaccined: Date;
    UniqueDose: boolean;
    Dose: number;
    Vaccines: Vaccine[];
    Pacients: Pacient[];
}